pkg/operator/ceph/object/objectstore.go

```diff
@@ -525,18 +526,30 @@ func createSystemUser(objContext *Context, namespace string) error {
 	return nil
 }
 
-func setMultisite(objContext *Context, store *cephv1.CephObjectStore, serviceIP string) error {
+func setMultisite(objContext *Context, store *cephv1.CephObjectStore) error {
 	logger.Debugf("setting multisite configuration for object-store %v", store.Name)
-	serviceEndpoint := fmt.Sprintf("http://%s:%d", serviceIP, store.Spec.Gateway.Port)
-	if store.Spec.Gateway.SecurePort != 0 {
-		serviceEndpoint = fmt.Sprintf("https://%s:%d", serviceIP, store.Spec.Gateway.SecurePort)
-	}
+	serviceEndpoint := objContext.Endpoint
 
 	if store.Spec.IsMultisite() {
+		if store.Spec.Zone.Endpoint != "" {
+			serviceEndpoint = store.Spec.Zone.Endpoint
+		}
 		zoneEndpointsList, err := getZoneEndpoints(objContext, serviceEndpoint)
 		if err != nil {
 			return err
 		}
+
+		// Remove old endpoint of RGW server from zone if it is different
+		if store.Status != nil && store.Status.Info != nil {
+			statusInfoEndpoint, isEndpointPresent := store.Status.Info["endpoint"]
+			if isEndpointPresent && statusInfoEndpoint != serviceEndpoint {
+				for i, endpoint := range zoneEndpointsList {
+					if store.Status.Info["endpoint"] == endpoint {
+						zoneEndpointsList = append(zoneEndpointsList[:i], zoneEndpointsList[i+1:]...)
+					}
+				}
+			}
+		}
```

Comment:
I share Travis's concerns here. Users can edit this manually via the toolbox, and we have examples of them doing just that to work around this issue.

Let's have this in a helper function that is tested, including corner cases like multiple endpoints and no endpoints.